### ISSUE TITLE
[BUG] `npm list -g` doesn't display a specific globally installed package, ESLint

### ISSUE THREAD

**Issue by darkred** [npm/cli#3877]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have ESLint 8.0.0 installed globally via  `npm install -g eslint` (using Node v16.11.0 x64/npm v8.0.0, as well as with npm7.x, on win11).

The issue is that, if I run `npm list -g` then unfortunately ESLint is not listed (it should show: `eslint@8.0.0`).

For reference, all my other globally installed packages are displayed ok:
```
> npm list -g  (or: npm list -g --depth=0)

C:\Users\Kostas\AppData\Roaming\npm
+-- npm-check-updates@11.8.5
+-- npm-gui@3.1.1
+-- prettier@2.4.1
+-- stylelint@13.13.1
+-- typescript@4.4.3
+-- web-ext@6.4.0
`-- zx@4.2.0
```
I have tried uninstalling+reinstalling ESLInt without any difference. I have also tried in a virtual machine (=clean install of Node) with same results.

But, with this command (using `ls` of 'Git for Windows' installed Bash), ESLint is included in the list:
```
> ls -l $(npm root -g)

    Directory: C:\Users\Kostas\AppData\Roaming\npm\node_modules

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----          12/10/2021    15:40                eslint
d----          12/10/2021    15:40                npm-check-updates
d----          12/10/2021    15:40                npm-gui
d----          12/10/2021    15:40                prettier
d----          12/10/2021    15:40                stylelint
d----          12/10/2021    15:40                typescript
d----          12/10/2021    15:40                web-ext
d----          12/10/2021    15:40                zx
```

### Expected Behavior

`npm list -g` should show: `eslint@8.0.0` .

### Steps To Reproduce

1. In a fresh Node x64 v16.11.0/npm v8.0.0 (or v16.10.0/npm v7.24.0) install ESLint via ``npm install -g eslint``
2. (no special config)
3. Run `npm list -g` (or `npm list -g --depth=0`)
4. Notice that ESLint is not listed.

### Environment

- OS: win 11 x64
- Node: 16.10.0/16.11.0
- npm: 7.24.0/8.0.0

**Comment by darkred** [comment]
The issue still occurs with Node 17.0.0 w/ npm 8.1.0 .

**Comment by tavrez** [comment]
I have 3 packages with this problem: eslint, semver, yarn

**Comment by MuTsunTsai** [comment]
Same problem here for eslint on Win 11 x64, Node 16.13.1, NPM 8.1.2

**Comment by DollarAkshay** [comment]
Same problem with `npm@8.3.1`

**Comment by Cloud7050** [comment]
Can confirm for `eslint` (`8.9.0`) on `npm@8.5.1`

**Comment by rustedwizard** [comment]
Exactly same issue here with eslint on npm@8.5.3 and nodejs@16.14.0. eslint install/uninstall as global package without issue.

**Comment by ljharb** [comment]
Those with this problem for eslint - what does `which eslint` say, as well as `npm root -g` and `npm config get prefix`?

**Comment by rustedwizard** [comment]
I am on Windows 11 and using Powershell, so the equivelent command to `which` would be `get-command` and gives following output:

```
CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
ExternalScript  eslint.ps1                                                    C:\Users\chens\AppData\Roaming\npm\eslin…
```

`npm root -g` gives following output:

`C:\Users\chens\AppData\Roaming\npm\node_modules`

`npm config get prefix` gives following output:

`C:\Users\chens\AppData\Roaming\npm`

**Comment by Cloud7050** [comment]
(I'm on Windows 11) `where eslint`:

> `C:\Users\x\AppData\Roaming\npm\eslint`
> `C:\Users\x\AppData\Roaming\npm\eslint.cmd`

`npm root -g`:

> `C:\Users\x\AppData\Roaming\npm\node_modules`

`npm config get prefix`:

> `C:\Users\x\AppData\Roaming\npm`

**Comment by ljharb** [comment]
@rustedwizard thanks! all of those conceptually agree, so for you at least, my hunch is wrong.

Is anyone seeing this issue either "where those three don't agree", or, on "not windows"?

**Comment by tavrez** [comment]
Seems like v8.5.5 fixed the issue for me, I can see `eslint` when I run `npm -g ls` now

**Comment by MuTsunTsai** [comment]
I can also confirm that npm 8.5.5 seems to fixed the issue at least for eslint.

**Comment by rustedwizard** [comment]
I can also confirm that this problem is no longer present for npm 8.5.5

**Comment by darkred** [comment]
I confirm the issue is solved after updating npm to 8.5.5 . I'm glad it's fixed.
